netket/optimizer/sr.py

```diff
@@ -24,6 +27,37 @@
 from .preconditioner import AbstractLinearPreconditioner
 
 
+def check_conflicting_args_in_partial(
+    qgt: functools.partial | Any,
+    conflicting_args: List[str],
+    warning_message: str,
+) -> None:
+    """
+    Check for conflicting arguments in a QGT partial.
+
+    Args:
+        qgt: The partial to perform the check or any other object.
+        conflicting_args: List of argument names to check for conflicts.
+        warning_message: Warning message for the user.
+
+    Raises:
+        UserWarning: If conflicting arguments are found.
+    """
+
+    if not isinstance(qgt, functools.partial):
+        return
+
+    specified_args = set(qgt.keywords.keys())
+    conflicting = set(conflicting_args).intersection(specified_args)
+    if conflicting:
+        warnings.warn(
+            warning_message.format(
+                conflicting, {k: qgt.keywords[k] for k in conflicting}
+            ),
+            UserWarning,
```

Comment:
```suggestion
            UserWarning,
            stacklevel=3,
```

stacklevel=1 (default) would make the warning appear as raised here.
stacklvel=2 make the warning appear as if thrown from the function calling check conflicting args...
stacklevel=3 makes the warning appear as if thrown from the user when he constructs SR(...)

so I think the latter makes sense here.